File: code-examples/cpp/faq/maxTimeMS-example.cpp
// Include necessary MongoDB C++ driver headers
#include <bsoncxx/builder/stream/document.hpp>
#include <mongocxx/client.hpp>
#include <mongocxx/stdx.hpp>
#include <mongocxx/uri.hpp>
#include <mongocxx/instance.hpp>

// Use MongoDB namespaces for convenience
using bsoncxx::builder::basic::kvp;
using bsoncxx::builder::basic::make_document;

int main() {
    mongocxx::instance inst{}; // This should be done only once
    mongocxx::client conn{mongocxx::uri{}};

    auto collection = conn["database"]["collection"];
    try {
        // Execute a find command with a custom condition and max execution time
        collection.find_with_opts(
            make_document(kvp("$where", "sleep(100) || true")),
            mongocxx::options::find{}.max_time(std::chrono::milliseconds(50))
        );
    } catch (const mongocxx::query_exception& ex) {
        // Here you can handle the timeout exception
    }
    return 0;
}

File: code-examples/php/faq/maxTimeMS-example.php
<?php
require 'vendor/autoload.php';

// Connect to MongoDB
$client = new MongoDB\Client("mongodb://localhost:27017");

// Choose the database and collection
$collection = $client->demo->beers;

// Create query with sleep command
$filter = ['where' => 'sleep(100) || true'];
$options = ['maxTimeMS' => 50];

// Execute find command
try {
    $result = $collection->find($filter, $options);
} catch (Exception $e) {
    echo 'Caught exception: ',  $e->getMessage(), "\n";
}
?>
